### PR TITLE
Added JavaDoc comments to few methods and classes

### PR DESCRIPTION
**Pull request by prajgov**
Added JavaDoc comments to few Java classes.

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/CommonsApplication.java [hunk 1]
```diff
@@ -44,21 +44,32 @@
...
+    /**
+     * Provides a way to get member refWatcher
+     *
+     * @param context Application context
+     * @return application member refWatcher
+     */
+    public static RefWatcher getRefWatcher(Context context) {
+        CommonsApplication application = (CommonsApplication) context.getApplicationContext();
+        return application.refWatcher;
+    }
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/CommonsApplication.java]: Any particular reason to have moved this method?

File: app/src/main/java/fr/free/nrw/commons/LicenseList.java [hunk 2]
```diff
@@ -13,10 +13,18 @@
 import java.util.Locale;
 import java.util.Map;
 
+/**
+ * Represents LicenseList
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/LicenseList.java]: A list of licenses, actually :-)
Would you mind correcting?

File: app/src/main/java/fr/free/nrw/commons/Media.java [hunk 3]
```diff
@@ -90,19 +109,33 @@ public Media(Parcel in) {
...
+     * Gets tag of media
+     * @param key Media key
+     * @return Media tag
+     */
     public Object getTag(String key) {
         return tags.get(key);
     }
 
+    /**
+     * Modifies tag of media
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/Media.java]: or create

File: app/src/main/java/fr/free/nrw/commons/CommonsApplication.java (hunk 1 above)
Comment (prajgov) [reply app/src/main/java/fr/free/nrw/commons/CommonsApplication.java]: Auto code clean up may have been the cause of it

File: app/src/main/java/fr/free/nrw/commons/LicenseList.java (hunk 2 above)
Comment (prajgov) [reply app/src/main/java/fr/free/nrw/commons/LicenseList.java]: No problem I will change it

File: app/src/main/java/fr/free/nrw/commons/CommonsApplication.java (hunk 1 above)
Comment (nicolas-raoul) [reply app/src/main/java/fr/free/nrw/commons/CommonsApplication.java]: Could you please move it back? It was intentionally just below the refWatcher declaration.
You can use the page https://github.com/commons-app/apps-android-commons/pull/1025/files to see exactly what you changed :-) Thanks!

File: app/src/main/java/fr/free/nrw/commons/LicenseList.java [hunk 4]
```diff
@@ -13,10 +13,18 @@
 import java.util.Locale;
 import java.util.Map;
 
+/**
+ * Represents a list of
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/LicenseList.java]: missing a word I guess?

File: app/src/main/java/fr/free/nrw/commons/Utils.java [hunk 5]
```diff
@@ -37,12 +37,23 @@ public static String stripLocalizedString(String s) {
...
+     * @return URL of thumbnail
+     */
     public static String makeThumbBaseUrl(@NonNull String filename) {
         String name = new PageTitle(filename).getPrefixedText();
         String sha = new String(Hex.encodeHex(DigestUtils.md5(name)));
         return String.format("%s/%s/%s/%s", BuildConfig.IMAGE_URL_BASE, sha.substring(0, 1), sha.substring(0, 2), urlEncode(name));
     }
 
+    /**
+     * Encodes given URL in UTF-8 format
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/Utils.java]: Tricky, please write "URLEncode an URL and make its first character uppercase".

File: app/src/main/java/fr/free/nrw/commons/Utils.java [hunk 6]
```diff
@@ -37,12 +37,23 @@ public static String stripLocalizedString(String s) {
...
+     * @param url Unformatted URL
+     * @return Formatted URL
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/Utils.java]: "Encoded URL"

File: app/src/main/java/fr/free/nrw/commons/WelcomeActivity.java [hunk 7]
```diff
@@ -18,6 +18,21 @@
 
     private WelcomePagerAdapter adapter = new WelcomePagerAdapter();
 
+    /**
+     * Creates a way to change current activity to WelcomeActivity
+     *
+     * @param context Activity context
+     */
+    public static void startYourself(Context context) {
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/WelcomeActivity.java]: Please move this method back where it was too, thanks!

File: app/src/main/java/fr/free/nrw/commons/WelcomePagerAdapter.java [hunk 8]
```diff
@@ -52,16 +63,32 @@ public Object instantiateItem(ViewGroup container, int position) {
...
+     * @param position Index of item
+     * @param obj Adapter object
+     */
     @Override
     public void destroyItem(ViewGroup container, int position, Object obj) {
         container.removeView((View) obj);
     }
 
+    /**
+     * Interface used create on clicked callback(s).
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/WelcomePagerAdapter.java]: I am actually not sure what this method does, so can you please remove the Javadoc? Thanks!

File: app/src/main/java/fr/free/nrw/commons/data/Category.java [hunk 9]
```diff
@@ -152,6 +100,112 @@ private static Category fromCursor(Cursor cursor) {
...
     }
 
+    /**
+     * Gets name
+     *
+     * @return name
+     */
+    public String getName() {
+        return name;
+    }
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/data/Category.java]: Please put all of these methods back where they were, thanks!

File: app/src/main/java/fr/free/nrw/commons/data/Category.java [hunk 10]
```diff
@@ -175,15 +229,30 @@ private static Category fromCursor(Cursor cursor) {
                 + COLUMN_TIMES_USED + " INTEGER"
                 + ");";
 
+        /**
+         * Creates new database with provided SQLite database
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/data/Category.java]: "Creates new table in provided SQLite database"

File: app/src/main/java/fr/free/nrw/commons/data/Category.java [hunk 11]
```diff
@@ -175,15 +229,30 @@ private static Category fromCursor(Cursor cursor) {
...
+         *
+         * @param db Category database
+         */
         public static void onCreate(SQLiteDatabase db) {
             db.execSQL(CREATE_TABLE_STATEMENT);
         }
 
+        /**
+         * Deletes existing database
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/data/Category.java]: Deletes existing table

File: app/src/main/java/fr/free/nrw/commons/mwapi/LogBuilder.java [hunk 12]
```diff
@@ -66,6 +84,9 @@ public void log(boolean force) {
         logTask.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR, this);
     }
 
+    /**
+     * Logs in  false
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/mwapi/LogBuilder.java]: Not sure what this does, so please remove this javadoc, thanks!

File: app/src/main/java/fr/free/nrw/commons/utils/LengthUtils.java [hunk 13]
```diff
@@ -37,25 +37,57 @@ public static double computeDistanceBetween(LatLng from, LatLng to) {
...
+     */
     private static double computeAngleBetween(LatLng from, LatLng to) {
         return distanceRadians(Math.toRadians(from.getLatitude()),
                 Math.toRadians(from.getLongitude()),
                 Math.toRadians(to.getLatitude()),
                 Math.toRadians(to.getLongitude()));
     }
 
+    /**
+     * Computes arc length between 2 points
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/utils/LengthUtils.java]: Good! :-)

File: app/src/main/java/fr/free/nrw/commons/WelcomePagerAdapter.java (hunk 8 above)
Comment (prajgov) [reply app/src/main/java/fr/free/nrw/commons/WelcomePagerAdapter.java]: Are mentioning that destroy item comment or the one for interface

File: app/src/main/java/fr/free/nrw/commons/utils/LengthUtils.java [hunk 14]
```diff
@@ -37,25 +37,57 @@ public static double computeDistanceBetween(LatLng from, LatLng to) {
...
+     * @param lat2 Latitude of point B
+     * @param lng2 Longitude of point B
+     * @return Arc length between the points
+     */
     private static double distanceRadians(double lat1, double lng1, double lat2, double lng2) {
         return arcHav(havDistance(lat1, lat2, lng1 - lng2));
     }
 
+    /**
+     * Gets twice the amount of arc sine of (x)^(1/2)
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/utils/LengthUtils.java]: "Calculate the inverse haversine of an angle. See https://en.wikipedia.org/wiki/Haversine_formula"

File: app/src/main/java/fr/free/nrw/commons/utils/LengthUtils.java [hunk 15]
```diff
@@ -37,25 +37,57 @@ public static double computeDistanceBetween(LatLng from, LatLng to) {
...
+     * @return Arc length between points
+     */
     private static double havDistance(double lat1, double lat2, double longitude) {
         return hav(lat1 - lat2) + hav(longitude) * Math.cos(lat1) * Math.cos(lat2);
     }
 
+    /**
+     * Computes square sine function for half of the value entered
+     * @param x Angle in radians
+     * @return One-forth of square of sine function
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/utils/LengthUtils.java]: Not exactly ;-) You can leave this one empty.

File: app/src/main/java/fr/free/nrw/commons/utils/LengthUtils.java [hunk 16]
```diff
@@ -37,25 +37,57 @@ public static double computeDistanceBetween(LatLng from, LatLng to) {
...
+     * @param lat2 Latitude of point B
+     * @param longitude Longitude on which they lie
+     * @return Arc length between points
+     */
     private static double havDistance(double lat1, double lat2, double longitude) {
         return hav(lat1 - lat2) + hav(longitude) * Math.cos(lat1) * Math.cos(lat2);
     }
 
+    /**
+     * Computes square sine function for half of the value entered
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/utils/LengthUtils.java]: "Compute haversine"

File: build.gradle [hunk 17]
```diff
@@ -7,7 +7,7 @@ buildscript {
         google()
     }
     dependencies {
-        classpath "com.android.tools.build:gradle:${project.gradleVersion}"
+        classpath 'com.android.tools.build:gradle:3.0.1'
```
Comment (nicolas-raoul) [inline build.gradle]: Is this intended?

File: app/src/main/java/fr/free/nrw/commons/Media.java [hunk 18]
```diff
@@ -90,19 +109,33 @@ public Media(Parcel in) {
...
+    /**
+     * Gets media display title
+     * @return Media title
+     */
     public String getDisplayTitle() {
         if (filename == null) {
             return "";
         }
-        // FIXME: Gross hack bercause my regex skills suck maybe or I am too lazy who knows
+        // FIXME: Gross hack because my regex skills suck maybe or I am too lazy who knows
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/Media.java]: Thanks!

File: app/src/main/java/fr/free/nrw/commons/WelcomePagerAdapter.java (hunk 8 above)
Comment (nicolas-raoul) [reply app/src/main/java/fr/free/nrw/commons/WelcomePagerAdapter.java]: You can remove the Javadoc of "public interface Callback {", thanks! :-)

**Pull request merged**